### PR TITLE
syncer: add 'reader pos' for task status 

### PR DESCRIPTION
**Pull request by knull-cn**
### What problem does this PR solve? 
query-status task to  find the sync-unit pos , was not realtime. it get from checkpoint,and update 30 second one time.
#1620 

### What is changed and how it works?
add one field for readerBinlog for event position, which dm.syncer get current event position

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has interface methods change

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: dm/proto/dmworker.proto [hunk 1]
```diff
@@ -132,12 +132,14 @@ message SyncStatus {
     int64 recentTps = 3;
     string masterBinlog = 4;
     string masterBinlogGtid = 5;
-    string syncerBinlog = 6;
-    string syncerBinlogGtid = 7;
-    repeated string blockingDDLs = 8; // sharding DDL which current is blocking
-    repeated ShardingGroup unresolvedGroups = 9; // sharding groups which current are un-resolved
-    bool synced = 10;  // whether sync is catched-up in this moment
-    string binlogType = 11;
+    string readerBinlog = 6;
```
Comment (lance6716) [inline dm/proto/dmworker.proto:135]: We may keep the existing order to keep compatibility 🤔

File: syncer/status.go [hunk 2]
```diff
@@ -31,18 +31,26 @@ func (s *Syncer) Status() interface{} {
 	totalTps := s.totalTps.Load()
 	tps := s.tps.Load()
 
+	s.currentLocationMu.Lock()
+	curLocaltion := s.currentLocationMu.currentLocation
```
Comment (lance6716) [inline syncer/status.go:35]:
```suggestion
	curLocation := s.currentLocationMu.currentLocation
```

File: syncer/status.go [hunk 3]
```diff
@@ -31,18 +31,26 @@ func (s *Syncer) Status() interface{} {
 	totalTps := s.totalTps.Load()
 	tps := s.tps.Load()
 
+	s.currentLocationMu.Lock()
```
Comment (lance6716) [inline syncer/status.go:34]: use RLock and RUnlock

File: dm/proto/dmworker.proto [hunk 4]
```diff
@@ -132,6 +132,8 @@ message SyncStatus {
     int64 recentTps = 3;
     string masterBinlog = 4;
     string masterBinlogGtid = 5;
+    string readerBinlog = 12;
+    string readerBinlogGtid = 13;
```
Comment (lichunzhu) [inline dm/proto/dmworker.proto:136]: Add these two lines after L142?

**Review by lichunzhu — APPROVED** [review]
Rest LGTM

**Comment by lance6716** [comment]
replaced by https://github.com/pingcap/dm/pull/1709 😛

**Pull request closed without merging**